### PR TITLE
Ban improvements

### PR DESCRIPTION
**Pull request by TheBigZet**
### Brief description of the changes:
Added two commands:

- /kill <name> - for faster killing
- /punish <name> <reason> - for killing and banning, so that the offender drops the eq

### Tested Changes:
- [ ] Needs testing with people.
- [X] Tested on random guy, should work

### CODE REVIEW

File: comfy_panel/admin.lua [hunk 1]
```diff
@@ -728,6 +728,54 @@ end
 
 comfy_panel_tabs['Admin'] = {gui = create_admin_panel, admin = true}
 
+commands.add_command("kill", "Kill a player", function(cmd)
+    local killer = game.get_player(cmd.player_index)
+    local victim = cmd.parameter
```
Comment (bits-orio) [inline comfy_panel/admin.lua:733]: I wouldn't call this variable `victim`. Rather just call it `suspect` :)

**Comment by TheBigZet** [comment]
Offender is now switched to his team before banning, so that the team can get their resources back in case of stealing etc.

**Review by mysticamber — CHANGES_REQUESTED** [review]

File: comfy_panel/admin.lua [hunk 2]
```diff
@@ -728,6 +729,56 @@ end
 
 comfy_panel_tabs['Admin'] = {gui = create_admin_panel, admin = true}
 
+commands.add_command("kill", "Kill a player", function(cmd)
```
Comment (mysticamber) [inline comfy_panel/admin.lua:732]: I'm hella picky, sorry. 

The second arg (Kill a player) is an explanation of the command. It's a bit self-explanatory, but I think it'd be good to include usage, Something like "Kill a player - /kill <player name>"

File: comfy_panel/admin.lua [hunk 3]
```diff
@@ -728,6 +729,56 @@ end
...
+	local killer = game.get_player(cmd.player_index)
+	local victim = cmd.parameter
+	if not killer.admin then
```
Comment (mysticamber) [inline comfy_panel/admin.lua:735]: Personally, I'd use If...elseif... else. That way you don't have to do 2 "returns" + it's clear that both the clauses have to be met.

File: comfy_panel/admin.lua [hunk 4]
```diff
@@ -728,6 +729,56 @@ end
...
+	for i in string.gmatch(cmd.parameter, '%S+') do t[#t + 1] = i end
+	local offender = t[1]
+	table.remove(t, 1)
+	message = table.concat(t, ' ')
+	if not game.players[offender] then
+		punisher.print("Invalid name", {r = 1, g = 0.5, b = 0.1})
+		return
+	end
+	offender = game.get_player(offender)
+	if string.len(message) < 5 then
```
Comment (mysticamber) [inline comfy_panel/admin.lua:768]: I think it's better if you put this higher, next to where message is declared(762). That way the offender code is next line 772.

File: comfy_panel/admin.lua [hunk 5]
```diff
@@ -728,6 +729,56 @@ end
...
+		punisher.print("No valid reason given, or reason is too short", {r = 1, g = 0.5, b = 0.1})
+		return
+	end
+	Server.to_discord_embed(offender.name .. " was banned by " .. punisher.name .. ". " .. "Reason: " .. message)
+	message = message .. " Appeal on discord. Link on biterbattles.org", {r = 1, g = 0.5, b = 0.1}
```
Comment (mysticamber) [inline comfy_panel/admin.lua:773]: What colour is this? Maybe better to use the presets using the colour library. The less hardcoded the better.

Comment (TheBigZet) [reply comfy_panel/admin.lua:773]: I think I've just copied it from some other command definition but ok :D

File: comfy_panel/admin.lua (hunk 3 above)
Comment (TheBigZet) [reply comfy_panel/admin.lua:735]: Is that what you mean?
`if killer.admin and game.players[victim] then`
`	victim = game.get_player(victim)`
`	kill(victim, killer)`
`elseif not killer.admin then`
`	killer.print("Only admins have licence for killing!")`
`elseif not game.players[victim] then`
`	killer.print("Invalid name")`
`end`

Comment (mysticamber) [reply comfy_panel/admin.lua:735]: use 3 back ticks to group the code comment together. Preserves format too
```
Like
    this
```

Back to the code.. 
Yes, but opposite, so you don't have to do the check twice. 

You just need to 
```
if killer.admin 
elseif not player or not player.valid
else 
end
```

I'd probably also just do

```
victim = game.get_player(victim)
```
then check `if not player or not player.valid`

Checking that player is valid, will limit this command to in game admins. If you think that Hedwig should eventually use it, see the existing trust command for how it was implemented.

**Review by mysticamber — CHANGES_REQUESTED** [review]

File: comfy_panel/admin.lua (hunk 3 above)
Comment (TheBigZet) [reply comfy_panel/admin.lua]: I'm not sure if that's what you had in mind, but I've changed it around

File: comfy_panel/admin.lua [hunk 6]
```diff
@@ -728,6 +729,50 @@ end
...
+			if offender.force.name == "spectator" then join_team(offender, global.chosen_team[offender.name], true) end -- switches offender to their team if he's spectating
+			kill(offender, punisher)
+			game.ban_player(offender, message)
+		elseif not offender.valid then
+			punisher.print("Invalid name", Color.warning)
+		else
+			punisher.print("No valid reason given, or reason is too short", Color.warning)
+		end
+	elseif not punisher.admin then
+		punisher.print("This is admin only command", Coloar.warning)
```
Comment (AwesomePatrol) [inline comfy_panel/admin.lua:769]:
```suggestion
		punisher.print("This is admin only command", Color.warning)
```

**Review by AwesomePatrol — CHANGES_REQUESTED** [review]
Otherwise lgtm

File: comfy_panel/admin.lua (hunk 3 above)
Comment (AwesomePatrol) [reply comfy_panel/admin.lua]: IMO this structure of `if` statements is much easier to read (you don't need to keep in mind what's the opposite of the last conditional and it avoids nesting). However, please don't change it back. This is also ok and if Amber likes it then it's for the better.

File: comfy_panel/admin.lua (hunk 6 above)
Comment (TheBigZet) [reply comfy_panel/admin.lua:769]: lol my bad

**Review by AwesomePatrol — APPROVED** [review]

**Review by mysticamber — APPROVED** [review]

**Pull request merged**